AirCasting/SearchAndFollow/SearchAndFollowMap/SearchAndFollowMap.swift

```diff
@@ -118,7 +132,34 @@ struct SearchAndFollowMap: UIViewRepresentable {
                 
             }
             parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
-            dotsPositionShouldChange(for: mapView)
+            startingPointHolder = parent.startingPoint
+            if parent.startingPointChanged {
+                startPositionShouldChange(for: mapView)
+                DispatchQueue.main.async { self.parent.startingPointChanged = false }
+            } else {
+                dotsPositionShouldChange(for: mapView)
+            }
+        }
+
+        func selectNewPointer(newid: Int, oldid: Int) {
+            DispatchQueue.main.async {
+                self.sessionSearched.forEach { m in
+                    guard let marker = m.userData as? Int else {
+                        Log.error("Unexpectedly found value other than Int in marker userData")
+                        assertionFailure()
+                        return
+                    }
+                    if marker == newid {
+                        let markerImage = UIImage(systemName: "circle.circle.fill")!.scalePreservingAspectRatio(targetSize: CGSize(width: 35, height: 35))
+                        let markerView = UIImageView(image: markerImage.withRenderingMode(.alwaysTemplate))
+                        m.iconView = markerView
+                    } else if marker == oldid {
+                        let markerImage = UIImage(systemName: "circle.circle.fill")!
+                        let markerView = UIImageView(image: markerImage.withRenderingMode(.alwaysTemplate))
+                        m.iconView = markerView
+                    }
```

Comment:
else is almost identical with `if` here. why don't you extract the
```Swift
let markerView = UIImageView(image: markerImage.withRenderingMode(.alwaysTemplate))
m.iconView = markerView
```
outside so it's not repeated?